packages/react-reconciler/src/ReactFiberCommitWork.js

```diff
@@ -2822,6 +2823,8 @@ function commitMutationEffectsOnFiber(
     }
     case HostRoot: {
       if (enableFloat && supportsResources) {
+        prepareToCommitHoistables();
```

Comment:
This is where the per commit cache gets reset. I considered having it done in render when calling `getResource` since you need to render to have a commit and we could avoid the extra API concept between Fiber and Host but that was also subtle and indirect so I ended up making it explicit